packages/twenty-server/src/engine/middlewares/middleware.service.ts

```
@@ -159,8 +163,6 @@ export class MiddlewareService {
       switch (error.code) {
         case AuthExceptionCode.UNAUTHENTICATED:
           return 401;
-        default:
```

I think this must have gone unnoticed when reviewed, or im missing something, but dont see why we would default to 400 here

---

Don't you think it's more dangerous not to have a default? Could be 5000 or 400 indeed. But now that might become a 200? Or did I miss someting?

---

That made a test fail I think ; but the test was wrong it should have return 401 or 403 not 400

---

Oh I understand why it was a 400 now, it's because it's during authentication and I suppose it's assuming that all custom exceptions thrown are forbiddenExceptions. I'll try to improve the logic because that does not seem safe, we could end up swallowing 5xx into 4xx if we throw a custom exception that reflects a 5xx